src/dune_pkg/opam_solver.ml

```diff
@@ -758,6 +758,26 @@ module Solver_result = struct
     }
 end
 
+let reachable pkgs ~roots =
```

Comment:
The `pkgs` argument maps package names to their dependencies. Can you rename it to reflect this.